### PR TITLE
[Merged by Bors] - feat(GroupTheory/GroupAction/Quotient): subgroup action orbit lemmas and equivs

### PR DESCRIPTION
**Pull request by jsm28**
Add some lemmas and equivs relating to orbits for and quotients by the action of a subgroup.

From AperiodicMonotilesLean.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/GroupTheory/GroupAction/Quotient.lean [hunk 1]
```diff
@@ -346,6 +346,91 @@ instance isPretransitive_quotient (G) [Group G] (H : Subgroup G) : IsPretransiti
 #align mul_action.is_pretransitive_quotient MulAction.isPretransitive_quotient
 #align add_action.is_pretransitive_quotient AddAction.isPretransitive_quotient
 
+variable {α}
+
+@[to_additive]
+lemma finite_quotient_of_pretransitive_of_finite_quotient [IsPretransitive α β] {H : Subgroup α}
```
Comment (riccardobrasca) [inline Mathlib/GroupTheory/GroupAction/Quotient.lean:352]: Why not `[Finite (α ⧸ H)]`? Also, can this be an instance?

File: Mathlib/GroupTheory/GroupAction/Quotient.lean [hunk 2]
```diff
@@ -346,6 +346,91 @@ instance isPretransitive_quotient (G) [Group G] (H : Subgroup G) : IsPretransiti
...
+    rfl
+
+variable (β)
+
+/-- A bijection between the orbits under the action of a subgroup `H` on `β`, and the orbits
+under the action of `H` on each orbit under the action of `G`. -/
+@[to_additive "A bijection between the orbits under the action of an additive subgroup `H` on `β`,
+and the orbits under the action of `H` on each orbit under the action of `G`."]
+noncomputable def equivSubgroupOrbits (H : Subgroup α) :
+    orbitRel.Quotient H β ≃ Σω : Ω, orbitRel.Quotient H (orbitRel.Quotient.orbit ω) := by
```
Comment (riccardobrasca) [inline Mathlib/GroupTheory/GroupAction/Quotient.lean:413]: Can you write it explicitly as a composition as `calc` block?

File: Mathlib/GroupTheory/GroupAction/Quotient.lean [hunk 3]
```diff
@@ -346,6 +346,91 @@ instance isPretransitive_quotient (G) [Group G] (H : Subgroup G) : IsPretransiti
...
+      (Subtype.val : Quotient.mk (orbitRel α β) ⁻¹' {ω} → β)) :=
+        (Setoid.sigmaQuotientEquivOfLe (orbitRel_subgroup_le H)).symm
+    _ ≃ Σω : Ω, orbitRel.Quotient H (orbitRel.Quotient.orbit ω) :=
+      Equiv.sigmaCongrRight fun ω ↦ (equivSubgroupOrbitsSetoidComap H ω).symm
+
+variable {β}
+
+@[to_additive]
+lemma finite_quotient_of_finite_quotient_of_finite_quotient {H : Subgroup α}
+    (hb : Finite (orbitRel.Quotient α β)) (h : Finite (α ⧸ H)) :
```
Comment (riccardobrasca) [inline Mathlib/GroupTheory/GroupAction/Quotient.lean:425]: Same as above.

File: Mathlib/GroupTheory/GroupAction/Quotient.lean (hunk 1 above)
Comment (jsm28) [reply Mathlib/GroupTheory/GroupAction/Quotient.lean]: Changed to an instance and to use instance argument.

File: Mathlib/GroupTheory/GroupAction/Quotient.lean (hunk 3 above)
Comment (jsm28) [reply Mathlib/GroupTheory/GroupAction/Quotient.lean]: Changed likewise.

**Comment by riccardobrasca** [comment]
Thanks!
bors merge

**Pull request closed without merging**